plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlJdbcConnectionCreation.java

```diff
@@ -87,9 +87,9 @@ public void testJdbcConnectionCreations()
         assertJdbcConnections("SELECT * FROM nation LIMIT 1", 3, Optional.empty());
         assertJdbcConnections("SELECT * FROM nation ORDER BY nationkey LIMIT 1", 3, Optional.empty());
         assertJdbcConnections("SELECT * FROM nation WHERE nationkey = 1", 3, Optional.empty());
-        assertJdbcConnections("SELECT avg(nationkey) FROM nation", 2, Optional.empty());
```

Comment:
This is a regression. Why do we now open more connections?

Reply:
Found out it is because I touched the code `PostgreSqlClient#readTableStatistics`, using the new api would open new connection, thus influence this test, would you want me to revert this change, `SqlServer` also the same case.

Reply:
reverted